### PR TITLE
chore:  add roadmap link in the footer

### LINKED ISSUE #2288: Add Link To Roadmap to WWW Footer
### Preliminary Checks

- [X] I have reviewed https://unkey.com/docs for existing features that would solve my problem

- [X] I have searched for existing feature requests: https://github.com/unkeyed/unkey/issues

- [X] This issue is not a question, general help request, or anything other than a feature request directly related to Unkey. Please ask questions in our Discord community: https://unkey.com/discord.

### Is your feature request related to a problem? Please describe.

Unkey has moved their roadmap to Github Discussions. In an effort to be more publicly visible and transparent with our roadmap, we want to an easily accessible link to it on the the Unkey website. 

### Describe the solution

Given a visitor to `www.unkey.com`: 
- when I scroll to the bottom of the page, I expect to see a link to `www.unkey.com/roadmap` under "Company."
- when I click the link, I expect to be redirected to `https://github.com/unkeyed/unkey/discussions/categories/roadmap`

**Technical details**: 
- should be a Next.js redirect like our Discord and Github links in the footer

### Describe alternatives you have considered (if any)

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by harshsbhat**
## What does this PR do?

image

Fixes #2288

Adds a roadmap link in the footer which redirects to `www.unkey.com/roadmap` and then `https://github.com/unkeyed/unkey/discussions/categories/roadmap` similar to discord.

https://github.com/unkeyed/unkey/pull/2281 probably did not run `pnpm fmt` that's why the extra changes in plain and feedback component.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Scroll down to the footer and click on the roadmap it should open the GitHub discussions in a new tab.

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [x] Ran `pnpm fmt`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Summary by CodeRabbit

- **New Features**
	- Updated error message in the Feedback component for clarity.
	- Added a "Roadmap" navigation link in the footer for easy access to the roadmap discussions.
	- Introduced a redirect function to guide users to the GitHub discussions page for the roadmap.

- **Bug Fixes**
	- Enhanced transaction handling and error management across various API functionalities, improving overall reliability and user experience.

### CODE REVIEW

File: apps/www/components/footer/footer.tsx [hunk 1]
```diff
@@ -13,6 +13,7 @@ const navigation = [
   {
     title: "Company",
     links: [
+      { title: "Roadmap", href: "/roadmap" },
```
Comment (mcstepp) [inline apps/www/components/footer/footer.tsx:16]: Let's move this link to after "Templates" 

About, Blog, Changelog, and Templates are definitely more important public company information than the roadmap

**Review by mcstepp — CHANGES_REQUESTED** [review]

**Comment by harshsbhat** [comment]
@mcstepp DONE

**Review by mcstepp — APPROVED** [review]
🚀

**Pull request merged**